distributions/aws/test/e2e/fixtures/cognito_dependencies.py

```diff
@@ -0,0 +1,249 @@
+    rand_name,
+    wait_for,
+    get_iam_client,
+    get_eks_client,
+    get_ec2_client,
+)
+from e2e.utils.config import configure_env_file
+from e2e.utils.custom_resources import get_ingress
+
+
+@pytest.fixture(scope="class")
+def cognito_bootstrap(
+    metadata, region, request, root_domain_name, root_domain_hosted_zone_id
+):
+    subdomain_name = rand_name("platform") + "." + root_domain_name
+
+    cognito_deps = {}
+
+    def on_create():
+        root_hosted_zone, subdomain_hosted_zone = create_subdomain_hosted_zone(
+            subdomain_name,
+            root_domain_name,
+            region,
+            root_domain_hosted_zone_id,
+        )
+        cognito_deps["route53"] = {
+            "rootDomain": {
+                "name": root_hosted_zone.domain,
+                "hostedZoneId": root_hosted_zone.id,
+            },
+            "subDomain": {
+                "name": subdomain_hosted_zone.domain,
+                "hostedZoneId": subdomain_hosted_zone.id,
+            },
+        }
+
+        (
+            root_certificate,
+            subdomain_cert_n_virginia,
+            subdomain_cert_deployment_region,
+        ) = create_certificates(region, subdomain_hosted_zone, root_hosted_zone)
+
+        cognito_deps["route53"]["rootDomain"]["certARN"] = root_certificate.arn
+
+        cognito_deps["route53"]["subDomain"][
+            "us-east-1-certARN"
+        ] = subdomain_cert_n_virginia.arn
+        cognito_deps["route53"]["subDomain"][
+            region + "-certARN"
+        ] = subdomain_cert_deployment_region.arn
+
+        userpool_name = subdomain_name
+        cognito_userpool, _ = create_cognito_userpool(
+            userpool_name,
+            region,
+            subdomain_hosted_zone,
+            subdomain_cert_n_virginia.arn,
+        )
+
+        cognito_deps["cognitoUserpool"] = {
+            "name": cognito_userpool.userpool_name,
+            "ARN": cognito_userpool.arn,
+            "appClientId": cognito_userpool.client_id,
+            "domain": cognito_userpool.userpool_domain,
+            "domainAliasTarget": cognito_userpool.cloudfront_domain,
+        }
+
+        cognito_deps["kubeflow"] = {"region": region}
+
+    def on_delete():
+        cfg = metadata.get("cognito_dependencies") or cognito_deps
+        delete_cognito_dependency_resources(cfg)
+
+    return configure_resource_fixture(
+        metadata, request, cognito_deps, "cognito_dependencies", on_create, on_delete
+    )
+
+
+@pytest.fixture(scope="class")
+def configure_kf_admin_role(metadata, region, request, account_id, cluster):
+    role_name = rand_name("kf-admin-role")
+    iam_client = get_iam_client(region)
+    iam_resource = boto3.resource("iam")
+
+    def on_create():
+        associate_iam_oidc_provider(cluster, region)
+        eks_client = get_eks_client(region)
+        response = eks_client.describe_cluster(name=cluster)
+
+        oidc_provider_url = response["cluster"]["identity"]["oidc"]["issuer"][8:]
```

Comment:
will this string split always be consistent at 8 chars in future?

Reply:
yes, we need to remove `https://` it has not changed since 2019 atleast